Adding block-attributes-override filter (#24)
infinum/eightshift-docs: docs/guides/extending-classes.md

---
id: extending-classes
title: Extending Classes
---

> We are providing abstract classes, interfaces, helpers, Dependency Injection and namespaces for your project. In general you can use anything as is from the vendors library as or you can modify/add functionality to the existing clases by extending it in your project.

Reviewer: ```suggestion
We are providing abstract classes, interfaces, helpers, dependency injection and namespaces for your project. In general, you can use anything as-is from the vendor library or you can modify/add the functionality of existing classes by extending them in your project.
```

Author: 👍

> If you are farmiliar with the extending classes (Object Inheritance) in php language then you can just skip this chapter but for the rest of you here is and awesome description on how advance stuff in PHP works:

Reviewer: ```suggestion
If you are familiar with the extending classes (object inheritance) in PHP language then you can just skip this chapter but for the rest of you here is an awesome description on how it works:
```

Author: 👍

* [Interfaces](https://www.php.net/manual/en/language.oop5.interfaces.php)
* [Dependency Injection](https://en.wikipedia.org/wiki/Dependency_injection)
* [Abstract Classes](https://www.php.net/manual/en/language.oop5.abstract.php)
* [Object Inheritance](https://www.php.net/manual/en/language.oop5.inheritance.php)
* [Namespaces](https://www.php.net/manual/en/language.namespaces.php)

> Please read this chapter before you continue because it will help you in the later chapters.

Reviewer: ```suggestion
Please get acquainted with these concepts before you continue because it will help you in the later chapters.
```

Author: 👍
